lib/bolt/plugin.rb

```diff
@@ -288,16 +299,24 @@ def resolve_single_reference(reference)
 
       validate_proc.call(reference)
 
-      begin
+      result = begin
         # Evaluate the plugin and then recursively evaluate any plugin returned by it.
         hook.call(reference)
       rescue StandardError => e
         loc = "resolve_reference in #{plugin_name}"
         raise PluginError::ExecutionError.new(e.message, plugin_name, loc)
       end
+
+      plugin_cache.write_cache(result) if cache?(reference)
+
+      result
     end
     private :resolve_single_reference
 
+    private def cache?(reference)
+      reference.key?('_cache') || @config.plugin_cache.key?('ttl')
```

Comment:
```suggestion
      reference.key?('_cache') || @config.plugin_cache.key?('ttl')
```

`Bolt::Config` doesn't receive config from `bolt-project.yaml` if that file only includes keys not found in `bolt.yaml`. In that case, `@config.plugin_cache` returns an empty hash, even if `plugin-cache` is configured.